### ISSUE TITLE
Populate from version 4.11.14

### ISSUE THREAD

**Issue by sekko27** [Automattic/mongoose#5685]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

If you specify field selection and population it will throw "Projection cannot have a mix of inclusion and exclusion." error. In older versions (before 4.11.14) it's executed properly.

**If the current behavior is a bug, please provide the steps to reproduce.**

```javascript
const Promise = require('bluebird');
const mongoose = require('mongoose');

mongoose.Promise = Promise;

const connection = mongoose.createConnection('mongodb://localhost:27017/mongoose-population');
connection.on('error', () => {
    throw new Error('Unable to connect')
});
connection.on('open', () => {
    const schemaA = new mongoose.Schema({
        name: String
    });
    const modelA = connection.model('A', schemaA);

    const schemaB = new mongoose.Schema({
        some: String,
        ref: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'A'
        }
    });
    const modelB = connection.model('B', schemaB);

    Promise.resolve(modelA.create({name: 'test'}))
        .then((a) => modelB.create({ref: a}))
        .then(() => {
            return modelB.find({}, {some: false, __v: false}).populate('ref').exec();
        })
        .then((result) => {
            console.log('OK', result);
            process.exit(0)
        })
        .catch((err) => {
            console.error(err.message);
            process.exit(1);
        })

});
```
**What is the expected behavior?**

It should run as before (because of minor version has been changed only).

**Please mention your node.js, mongoose and MongoDB version.**

* node - v6.11.3
* mongoose - v4.11.14
* mongodb - v3.4.9

**Comment by vkarpov15** [comment]
Looks like an issue with the fix for #5669 . Will fix asap, thanks for reporting :+1:

**Comment by vkarpov15** [comment]
As a workaround, do:

```javascript
return modelB.find({}, {some: 0, __v: 0}).populate('ref').exec();
```

There's a bug with how mongoose handles `false` in projections that I'll push a fix for shortly
